test/distributed/test_c10d_nccl.py

```diff
@@ -3529,6 +3529,61 @@ def test_new_group_local_sync_sanity_check(self):
     def test_new_group_local_sync_duplicated_pg(self):
         self._test_new_group_local_sync_duplicate_pg(backend="nccl")
 
+    @requires_nccl()
+    @skip_if_lt_x_gpu(4)
+    def test_gather_subgroup(self):
+        if self.rank > 3:
+            # just easier to write the test for exactly 4 gpus
+            return
+
+        store = c10d.FileStore(self.file_name, self.world_size)
+        torch.distributed.init_process_group(backend="nccl", store=store, rank=self.rank, world_size=self.world_size)
```

Comment:
on devgpu with 8 gpus (world_size=8), seems only rank 0...3 calls ``torch.distributed.init_process_group`` ? PG does not complain about no init from rank 4...7 ?

Reply:
good point. since I chose to make my test run with exactly 4 gpus for simplicity, i should not rely on self.world_size to report 4 in case someone changes it.  I will update this to pass 4 into init_process_group